### PR TITLE
bpo-30166: Import command-line parsing modules only when needed.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

File: Lib/code.py [hunk 1]
```diff
@@ -303,6 +302,7 @@ def interact(banner=None, readfunc=None, local=None, exitmsg=None):
 
 
 if __name__ == "__main__":
+    import argparse
```
Comment (berkerpeksag) [inline Lib/code.py:305]: Style nit: You may want to add a new line after `import argparse` like the other cases.

**Review by berkerpeksag — APPROVED** [review]
I'm not familiar with IDLE, but rest of the PR looks good to me. Just left a comment about code styling.

Thanks!

File: Lib/code.py (hunk 1 above)
Comment (serhiy-storchaka) [reply Lib/code.py:305]: This is a thing that compelled me to hesitate. Some usages of argparse have an empty line after imports, some doesn't have (and the majority doesn't have). I added a new line if the following code is large or separated on sections by empty lines, and didn't add it if the following code is short and dense, as in this case.

**Comment by serhiy-storchaka** [comment]
I left reviewing IDLE on @terryjreedy.

File: Lib/idlelib/pyshell.py [hunk 2]
```diff
@@ -31,14 +30,13 @@
 import tokenize
 import warnings
 
-from idlelib import testing  # bool value
+import idlelib
```
Comment (terryjreedy) [inline Lib/idlelib/pyshell.py]: Move this also to __main__.  'idlelib' by itself is never used in main body of module.

File: Lib/profile.py [hunk 3]
```diff
@@ -25,10 +25,8 @@
 
 
 import sys
-import os
```
Comment (terryjreedy) [inline Lib/profile.py:28]: def trace_dispatch (line 184 in 3.6.1) is preceded by an apparently out-of-date comment
    # Heavily optimized dispatch routine for os.times() timer
That os.times is no longer used is why the os import can be moved.
self.timer is now  time.process time

**Review by terryjreedy — CHANGES_REQUESTED** [review]
Moving imports not needed in the main body is consistent with what I did in Fall 2015.  It is also how I introduced a bug.  So I checked that os can be moved in profile.py and verified the changes in idlelib/pyshell.py.  (They will be helpful if I ever move main() to a different module.)  The one requested change is obviously not crucial.

**Comment by terryjreedy** [comment]
I have backported the idlelib.pyshell part to 3.6

**Review by ibanezz1999 — COMMENTED** [review]
Bahasa indonesia

**Comment by terryjreedy** [comment]
@ibanezz1999 Please explain your comment.  This was merged months ago and any additional changes will be a new patch.

**Pull request merged**